air-quality-ui/src/components/single-city/station-map/StationMap.spec.tsx

```diff
@@ -12,24 +13,173 @@ jest.mock('../../../context', () => ({
   }),
 }))
 
+const mockAddTo = jest.fn()
+const mockLngLat = jest.fn()
+const mockMarkerRemove = jest.fn()
 jest.mock('maplibre-gl', () => ({
-  Marker: jest.fn(),
+  Marker: jest.fn().mockImplementation(() => {
+    return {
+      setLngLat: mockLngLat,
+      addTo: mockAddTo,
+      setPopup: jest.fn(),
+      remove: mockMarkerRemove,
+      getLngLat: jest.fn(() => ({ lat: 1, lng: 2 })),
+      _color: '#123456',
+    }
+  }),
+  Popup: jest.fn(() => ({
+    setDOMContent: jest.fn(),
+  })),
   Map: jest.fn(() => ({
     addControl: jest.fn(),
   })),
   FullscreenControl: jest.fn(),
 }))
 
 describe('StationMap', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
   it('renders', async () => {
     render(
       <StationMap
         mapCenter={{ latitude: 1, longitude: 2 }}
-        locations={new Map()}
+        stations={{}}
+        visibleLocations={['']}
+        addSite={() => {}}
+        removeSite={() => {}}
+        stationColors={{}}
       />,
     )
     await waitFor(() => {
       expect(screen.queryByTestId('map')).toBeInTheDocument()
     })
   })
+
+  it('marker created correctly', async () => {
+    render(
+      <StationMap
+        mapCenter={{ latitude: 1, longitude: 2 }}
+        stations={{
+          stationA: { name: 'stationA', latitude: 1, longitude: 2 },
+        }}
+        visibleLocations={['stationA']}
+        addSite={() => {}}
+        removeSite={() => {}}
+        stationColors={{ stationA: '#123456' }}
+      />,
+    )
+    expect(libre.Marker).toHaveBeenCalledTimes(1)
+    expect(libre.Marker).toHaveBeenCalledWith({ color: '#123456' })
+    expect(mockAddTo).toHaveBeenCalledTimes(1)
+  })
+
+  it('marker updated correctly', async () => {
+    const visibleLocations = ['stationA']
+    const { container } = render(
+      <StationMap
+        mapCenter={{ latitude: 1, longitude: 2 }}
+        stations={{
+          stationA: { name: 'stationA', latitude: 1, longitude: 2 },
+        }}
+        visibleLocations={visibleLocations}
+        addSite={() => {}}
+        removeSite={() => {}}
+        stationColors={{ stationA: '#123456' }}
+      />,
+    )
+    expect(libre.Marker).toHaveBeenCalledTimes(1)
+    expect(libre.Marker).toHaveBeenCalledWith({ color: '#123456' })
+    expect(mockAddTo).toHaveBeenCalledTimes(1)
+
+    render(
```

Comment:
Rather than re-rendering can't we just remove the station from the visible stations prop to trigger the effect of a station being removed?
We could then add it back in to trigger the re-add too?

We could also pass in some functions to the addSite and removeSite props (might be a separate test) and call these to ensure the props drive the changes as well.